app/src/main/java/com/jerboa/Utils.kt

```diff
@@ -1569,6 +1634,12 @@ fun canMod(
     }
 }
 
+fun futureDaysToUnixTime(days: Int?): Int? {
+    return days?.let {
+        (Date.from(Instant.now().plus(Duration.ofDays(it.toLong()))).time / 1000L).toInt()
```

Comment:
Seems overly complex

Why not `Instant.now().plus(3, DAYS).toEpochMilli()` ?

Reply:
Thx, looks like this one works:

`Instant.now().plus(it.toLong(), ChronoUnit.DAYS).epochSecond.toInt()`

Reply:
There is reason that is a long, this will overflow in 2038

Reply:
👀 Then we'll also need to change that in the api-client also. Rust uses an `i64` so it should be fine there.

Reply:
api-client? lemmy-js-client? thats not gonna be a problem since its a float which has a much bigger range and will only start to lose precision in about 300k years

Reply:
https://github.com/MV-GH/LemmyBackwardsCompatibleAPI/blob/master/app/src/commonMain/kotlin/it/vercruysse/lemmyapi/v0x18/datatypes/BanPerson.kt#L11 and the same for `BanPersonFromCommunity`

Those could be `Long` , but if you're auto-generating them, we might need to figure out a smart way to do it, because javascript had too many issues with `BigInt` when we tried to do that the first time.

Reply:
Hmm that is annoying, I am guessing what ever tool you use to generate those TS types can't annotate its actual type for numbers?

I'll keep a manual list for now

Reply:
I commented over on your recent pr, but imo you might as well just replace every int with long.